services/src/main/java/com/linkedin/kmf/common/Utils.java

```diff
@@ -35,13 +54,178 @@
-    zkUtils.close();
-    return partition;
+    ZkUtils zkUtils = ZkUtils.apply(zkUrl, ZK_SESSION_TIMEOUT_MS, ZK_CONNECTION_TIMEOUT_MS, JaasUtils.isZkSecurityEnabled());
+    try {
+      Seq<String> topics = scala.collection.JavaConversions.asScalaBuffer(Arrays.asList(topic));
+      return zkUtils.getPartitionsForTopics(topics).apply(topic).size();
+    } finally {
+      zkUtils.close();
+    }
+  }
+
+  /**
+   * Create the topic that the monitor uses to monitor the cluster.  This method attempts to create a topic so that all
+   * the brokers in the cluster will have partitionFactor partitions.  If the topic exists, but has different parameters
+   * then this does nothing to update the parameters.
+   *
+   * TODO: Do we care about rack aware mode?  I would think no because we want to spread the topic over all brokers.
+   * @param zkUrl zookeeper connection url
+   * @param topic topic name
+   * @param replicationFactor the replication factor for the topic
+   * @param partitionFactor This is multiplied by the number brokers to compute the number of partitions in the topic.
+   * @return the number of partitions created
+   */
+  public static int createMonitoringTopicIfNotExists(String zkUrl, String topic, int replicationFactor,
+      int partitionFactor) {
+    ZkUtils zkUtils = ZkUtils.apply(zkUrl, ZK_SESSION_TIMEOUT_MS, ZK_CONNECTION_TIMEOUT_MS, JaasUtils.isZkSecurityEnabled());
+    try {
+      if (AdminUtils.topicExists(zkUtils, topic)) {
+        LOG.info("Monitoring topic \"" + topic + "\" already exists.");
+        return getPartitionNumForTopic(zkUrl, topic);
+      }
+
+      int brokerCount = zkUtils.getAllBrokersInCluster().size();
+
+      if (partitionFactor <= 0) {
+        throw new IllegalArgumentException("Partition factor must be greater than zero, but was configured for " +
+            partitionFactor + ".");
+      }
+      int partitionCount = brokerCount * partitionFactor;
+
+      int minIsr = Math.max(replicationFactor - 1, 1);
+      Properties topicConfig = new Properties();
+      topicConfig.setProperty(KafkaConfig.MinInSyncReplicasProp(), Integer.toString(minIsr));
+      AdminUtils.createTopic(zkUtils, topic, partitionCount, replicationFactor, new Properties());
+
+      LOG.info("Created monitoring topic \"" + topic + "\" with " + partitionCount + " partitions, min ISR of " + minIsr
+          + " and replication factor of " + replicationFactor + ".");
+
+      return partitionCount;
+    } finally {
+      zkUtils.close();
+    }
+  }
+
+  /**
+   * This is so you can call KafkaConsumer.partitionsFor don't consume messages from this that is the job of the
+   * ConsumeService.
+   *
+   * @param brokerList probably get this from the configuration file.
+   * @param zkUrl zoo keeper url
+   * @param topic the monitoring topic
+   * @return an initialized KafkaConsumer
+   */
+
+  private static KafkaConsumer<?,?> constructClientForMetadata(String brokerList, String zkUrl, String topic) {
```

Comment:
This code is related to partition rebalance. Do you intend to get these reviewed in this pull request or are they committed here by accident?